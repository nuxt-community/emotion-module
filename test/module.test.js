jest.setTimeout(60000)

const { Nuxt, Builder } = require('nuxt-edge')
const request = require('request-promise-native')
const getPort = require('get-port')

const config = require('../example/nuxt.config')
config.dev = false

let nuxt, port

const url = path => `http://localhost:${port}${path}`
const get = path => request(url(path))

describe('basic', () => {
  beforeAll(async () => {
    nuxt = new Nuxt(config)
    await nuxt.ready()
    await new Builder(nuxt).build()
    port = await getPort()
    await nuxt.listen(port)
  })

  afterAll(async () => {
    await nuxt.close()
  })

  test('/ (pink)', async () => {
    const html = await get('/')
    expect(html).toMatchSnapshot()
  })

  test('/ (normal)', async () => {
    const html = await get('/normal')
    expect(html).toMatchSnapshot()
  })
})
