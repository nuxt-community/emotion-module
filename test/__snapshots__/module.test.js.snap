// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module / (normal) 1`] = `
"<!doctype html>
<html data-n-head-ssr>
  <head >
    <style data-vue-ssr-id=\\"124e288c:0\\">.nuxt-progress{position:fixed;top:0;left:0;right:0;height:2px;width:0;opacity:1;transition:width .1s,opacity .4s;background-color:#000;z-index:999999}.nuxt-progress.nuxt-progress-notransition{transition:none}.nuxt-progress-failed{background-color:red}</style><style>@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro');html,body{font-family:'Source Sans Pro',sans-serif;color:#0a1f44;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:100vh;text-align:center;}a{padding:.5em;border:1px solid grey;border-radius:.3em;color:#0a1f44;-webkit-text-decoration:none;text-decoration:none;}.css-1g3y0td{color:#42b883;}.css-1riqxhg{font-size:60px;}</style><script>window.$emotionSSRIds = [\\"yhjl76\\",\\"1g3y0td\\",\\"1riqxhg\\"]</script>
  </head>
  <body >
    <div data-server-rendered=\\"true\\" id=\\"__nuxt\\"><!----><div id=\\"__layout\\"><div><div><h1 class=\\"css-1riqxhg\\">
    Nuxt
    <span class=\\"css-1g3y0td\\">
      JS
    </span></h1> <h1>
    The
    <span class=\\"css-1g3y0td\\">
      Vue.js
    </span>
    Framework
  </h1></div> <a href=\\"/\\" class=\\"nuxt-link-active\\">
    Go Pink
  </a></div></div></div><script>window.__NUXT__={layout:\\"default\\",data:[{}],fetch:[],error:null,serverRendered:true,routePath:\\"\\\\u002Fnormal\\",config:{}};</script><script src=\\"/_nuxt/runtime.js\\" defer></script><script src=\\"/_nuxt/pages/normal.js\\" defer></script><script src=\\"/_nuxt/commons.app.js\\" defer></script><script src=\\"/_nuxt/vendors.app.js\\" defer></script><script src=\\"/_nuxt/app.js\\" defer></script>
  </body>
</html>
"
`;

exports[`module / (pink) 1`] = `
"<!doctype html>
<html data-n-head-ssr>
  <head >
    <style data-vue-ssr-id=\\"124e288c:0\\">.nuxt-progress{position:fixed;top:0;left:0;right:0;height:2px;width:0;opacity:1;transition:width .1s,opacity .4s;background-color:#000;z-index:999999}.nuxt-progress.nuxt-progress-notransition{transition:none}.nuxt-progress-failed{background-color:red}</style><style>@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro');html,body{font-family:'Source Sans Pro',sans-serif;color:#0a1f44;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;min-height:100vh;text-align:center;}a{padding:.5em;border:1px solid grey;border-radius:.3em;color:#0a1f44;-webkit-text-decoration:none;text-decoration:none;}.css-1g3y0td{color:#42b883;}.css-1riqxhg{font-size:60px;}.css-3sn2xs{color:hotpink;}</style><script>window.$emotionSSRIds = [\\"yhjl76\\",\\"1g3y0td\\",\\"1riqxhg\\",\\"3sn2xs\\"]</script>
  </head>
  <body >
    <div data-server-rendered=\\"true\\" id=\\"__nuxt\\"><!----><div id=\\"__layout\\"><div class=\\"css-3sn2xs\\"><div><h1 class=\\"css-1riqxhg\\">
    Nuxt
    <span class=\\"css-1g3y0td\\">
      JS
    </span></h1> <h1>
    The
    <span class=\\"css-1g3y0td\\">
      Vue.js
    </span>
    Framework
  </h1></div> <a href=\\"/normal\\">
    Go Normal
  </a></div></div></div><script>window.__NUXT__={layout:\\"default\\",data:[{}],fetch:[],error:null,serverRendered:true,routePath:\\"\\\\u002F\\",config:{}};</script><script src=\\"/_nuxt/runtime.js\\" defer></script><script src=\\"/_nuxt/pages/index.js\\" defer></script><script src=\\"/_nuxt/commons.app.js\\" defer></script><script src=\\"/_nuxt/vendors.app.js\\" defer></script><script src=\\"/_nuxt/app.js\\" defer></script>
  </body>
</html>
"
`;
